packages/boot/src/mixins/boot.mixin.ts

```diff
@@ -134,9 +141,11 @@ export function _bindBooter(
   ctx: Context,
   booterCls: Constructor<Booter>,
 ): Binding {
-  return ctx
-    .bind(`${BootBindings.BOOTER_PREFIX}.${booterCls.name}`)
-    .toClass(booterCls)
-    .inScope(BindingScope.CONTEXT)
-    .tag(BootBindings.BOOTER_TAG);
+  const binding = createBindingFromClass(booterCls, {
+    namespace: BootBindings.BOOTER_PREFIX,
+  })
+    .tag(BootBindings.BOOTER_TAG)
+    .inScope(BindingScope.SINGLETON);
```

Comment:
Hmm, this is making multiple changes in a way that makes it difficult to spot them:
- change the scope from CONTEXT to SINGLETON
- start honouring `@bind` decorator by calling `createBindingFromClass`, at least I think that's the case?

Let's split these changes into two commits please, if not two pull requests.

What's more important, please add tests to verify these changes and prevent regressions in the future.

It would be ideal to have an acceptance-level test that fails when a booter is bound in CONTEXT scope.

Reply:
Split the PR into two commits and added corresponding tests.